### PR TITLE
Preemptively override s.c.i.HashMap#updatedWith 

### PR DESCRIPTION
**Pull request by joshlemer**
This is in anticipation of future performance optimizations which will target 2.13.1, which currently can be found [here](https://github.com/scala/scala/pull/8032)

### CODE REVIEW

**Review by NthPortal — APPROVED** [review]

**Comment by NthPortal** [comment]
looks like MiMa is not happy with it - I think you need to either add an exception (`ProblemFilters.exclude[IncompatibleResultTypeProblem]("scala.collection.immutable.HashMap.updatedWith")`) or wait until the PR disabling MiMa gets merged

**Review by lrytz — APPROVED** [review]

**Pull request merged**
